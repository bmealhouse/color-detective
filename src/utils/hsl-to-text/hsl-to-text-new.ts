import { Hsl } from "..";
import { hslColors, gray, hslColorConfig } from "./hsl-colors";

export function hslToTextNew({ h, s, l }: Hsl): string {
  // regardless of hue, lightness < 5 should classify as black
  if (l < 10) {
    return "black";
  }

  // regardless of hue, lightness >= 85 should classify as white
  if (l >= 85) {
    return "white";
  }

  if (s < 10) {
    if (l < 25) return "dark gray";
    if (l < 35) return "todo";
    if (l >= 75) return "white";
    if (l >= 65) return "light gray";
    return "gray";
  }

  const colorIndex = Math.floor((h % 360) / 10);

  // let previousColorIndex = colorIndex - 1;
  // if (previousColorIndex < 0) {
  //   previousColorIndex = hslColorConfig.length;
  // }

  // let nextColorIndex = colorIndex + 1;
  // if (nextColorIndex > hslColorConfig.length) {
  //   nextColorIndex = 0;
  // }

  // const previousColor = hslColorConfig[previousColorIndex];
  const {
    color,
    darkColor,
    darkerColor,
    lightColor,
    lighterColor,
    lowSaturation,
    mediumSaturation,
    highSaturation,
  } = hslColorConfig[colorIndex];
  // const nextColor = hslColorConfig[nextColorIndex];

  if (l < 25) {
    if (s < 35) {
      return lowSaturation.darkerColor;
    }
    if (s < 60) {
      return mediumSaturation.darkerColor;
    }
    if (s < 85) {
      return highSaturation.darkerColor;
    }
    return darkerColor;
  }

  if (l < 35) {
    if (s < 35) {
      return lowSaturation.darkColor;
    }
    if (s < 60) {
      return mediumSaturation.darkColor;
    }
    if (s < 85) {
      return highSaturation.darkColor;
    }
    return darkColor;
  }

  if (l >= 75) {
    if (s < 35) {
      return lowSaturation.lighterColor;
    }
    if (s < 60) {
      return mediumSaturation.lighterColor;
    }
    if (s < 85) {
      return highSaturation.lighterColor;
    }
    return lighterColor;
  }

  if (l >= 65) {
    if (s < 35) {
      return lowSaturation.lightColor;
    }
    if (s < 60) {
      return mediumSaturation.lightColor;
    }
    if (s < 85) {
      return highSaturation.lightColor;
    }
    return lightColor;
  }

  if (s < 35) {
    return lowSaturation.color;
  }
  if (s < 60) {
    return mediumSaturation.color;
  }
  if (s < 85) {
    return highSaturation.color;
  }

  return color;

  // for (const colorInfo of hslColors) {
  //   const {
  //     colorName,
  //     upperBound,
  //     thresholds,
  //     alternateColorNames,
  //   } = colorInfo;

  //   if (h % 360 < upperBound) {
  //     if (l < thresholds.black) {
  //       return "black";
  //     }

  //     if (l >= thresholds.white) {
  //       return "white";
  //     }

  //     // TODO: handle hue variance
  //     // TODO: handle saturation variance

  //     if (s <= 25) {
  //       return alternateColorNames.light;
  //     } else if (s <= 60 && l >= 60) {
  //       return alternateColorNames.light;
  //     }

  //     if (l < thresholds.dark) {
  //       return alternateColorNames.dark;
  //     }

  //     if (l >= thresholds.veryLight) {
  //       return alternateColorNames.veryLight;
  //     }

  //     if (l >= thresholds.light) {
  //       return alternateColorNames.light;
  //     }

  //     return colorName;
  //   }
  // }

  // return "unknown";
}

// export function hslToTextNew({ h, s, l }: Hsl): string {
//   if (l < 5) {
//     return "black";
//   }

//   if (l >= 90) {
//     return "white";
//   }

//   if (s <= 10) {
//     if (l < gray.thresholds.black) {
//       return "black";
//     }
//     if (l >= gray.thresholds.white) {
//       return "white";
//     }
//     return "gray";
//   }

//   for (const colorInfo of hslColors) {
//     const {
//       colorName,
//       upperBound,
//       thresholds,
//       alternateColorNames,
//     } = colorInfo;

//     if (h % 360 < upperBound) {
//       if (l < thresholds.black) {
//         return "black";
//       }

//       if (l >= thresholds.white) {
//         return "white";
//       }

//       // TODO: handle hue variance
//       // TODO: handle saturation variance

//       if (s <= 25) {
//         return alternateColorNames.light;
//       } else if (s <= 60 && l >= 60) {
//         return alternateColorNames.light;
//       }

//       if (l < thresholds.dark) {
//         return alternateColorNames.dark;
//       }

//       if (l >= thresholds.veryLight) {
//         return alternateColorNames.veryLight;
//       }

//       if (l >= thresholds.light) {
//         return alternateColorNames.light;
//       }

//       return colorName;
//     }
//   }

//   return "unknown";
// }

// COLOR      DARK COLOR      LIGHT COLOR
// black      black           gray
// blue       dark blue       light blue
// brown      brown           tan
// gray       gray            gray
// green      dark green      light green
// orange     orange          peach
// pink       dark pink       light pink
// purple     purple          light purple
// red        red             pink
// teal       dark teal       light teal
// white      gray            white
// yellow     dark yellow     yellow
