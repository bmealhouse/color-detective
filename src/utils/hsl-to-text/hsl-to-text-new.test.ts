import { hslToTextNew as hslToText } from "./hsl-to-text-new";

const hue = [0]; // multiplesOf(10).until(360);
const saturation = multiplesOf(5).until(100);
const lightness = multiplesOf(5).until(85);

function multiplesOf(multiple: number) {
  return {
    until(max: number) {
      return Array.from(Array((max + multiple) / multiple).keys()).map(
        (x) => x * multiple
      );
    },
  };
}

hue.forEach((h) => {
  saturation.forEach((s) => {
    lightness.forEach((l) => {
      test(`hsl(${h}, ${s}%, ${l}%)`, () => {
        expect(hslToText({ h, s, l })).toMatchSnapshot();
      });
    });
  });
});

test("hue values to test are in increments of 5", () => {
  expect(hue).toMatchInlineSnapshot(`
    Array [
      0,
    ]
  `);
});

test("saturation values to test are in increments of 5", () => {
  expect(saturation).toMatchInlineSnapshot(`
    Array [
      0,
      5,
      10,
      15,
      20,
      25,
      30,
      35,
      40,
      45,
      50,
      55,
      60,
      65,
      70,
      75,
      80,
      85,
      90,
      95,
      100,
    ]
  `);
});

test("lightness values to test are in increments of 5", () => {
  expect(lightness).toMatchInlineSnapshot(`
    Array [
      0,
      5,
      10,
      15,
      20,
      25,
      30,
      35,
      40,
      45,
      50,
      55,
      60,
      65,
      70,
      75,
      80,
      85,
    ]
  `);
});
