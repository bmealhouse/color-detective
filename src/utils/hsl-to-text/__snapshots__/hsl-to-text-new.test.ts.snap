// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hsl(0, 0%, 0%) 1`] = `"black"`;

exports[`hsl(0, 0%, 5%) 1`] = `"black"`;

exports[`hsl(0, 0%, 10%) 1`] = `"dark gray"`;

exports[`hsl(0, 0%, 15%) 1`] = `"dark gray"`;

exports[`hsl(0, 0%, 20%) 1`] = `"dark gray"`;

exports[`hsl(0, 0%, 25%) 1`] = `"todo"`;

exports[`hsl(0, 0%, 30%) 1`] = `"todo"`;

exports[`hsl(0, 0%, 35%) 1`] = `"gray"`;

exports[`hsl(0, 0%, 40%) 1`] = `"gray"`;

exports[`hsl(0, 0%, 45%) 1`] = `"gray"`;

exports[`hsl(0, 0%, 50%) 1`] = `"gray"`;

exports[`hsl(0, 0%, 55%) 1`] = `"gray"`;

exports[`hsl(0, 0%, 60%) 1`] = `"gray"`;

exports[`hsl(0, 0%, 65%) 1`] = `"light gray"`;

exports[`hsl(0, 0%, 70%) 1`] = `"light gray"`;

exports[`hsl(0, 0%, 75%) 1`] = `"white"`;

exports[`hsl(0, 0%, 80%) 1`] = `"white"`;

exports[`hsl(0, 0%, 85%) 1`] = `"white"`;

exports[`hsl(0, 5%, 0%) 1`] = `"black"`;

exports[`hsl(0, 5%, 5%) 1`] = `"black"`;

exports[`hsl(0, 5%, 10%) 1`] = `"dark gray"`;

exports[`hsl(0, 5%, 15%) 1`] = `"dark gray"`;

exports[`hsl(0, 5%, 20%) 1`] = `"dark gray"`;

exports[`hsl(0, 5%, 25%) 1`] = `"todo"`;

exports[`hsl(0, 5%, 30%) 1`] = `"todo"`;

exports[`hsl(0, 5%, 35%) 1`] = `"gray"`;

exports[`hsl(0, 5%, 40%) 1`] = `"gray"`;

exports[`hsl(0, 5%, 45%) 1`] = `"gray"`;

exports[`hsl(0, 5%, 50%) 1`] = `"gray"`;

exports[`hsl(0, 5%, 55%) 1`] = `"gray"`;

exports[`hsl(0, 5%, 60%) 1`] = `"gray"`;

exports[`hsl(0, 5%, 65%) 1`] = `"light gray"`;

exports[`hsl(0, 5%, 70%) 1`] = `"light gray"`;

exports[`hsl(0, 5%, 75%) 1`] = `"white"`;

exports[`hsl(0, 5%, 80%) 1`] = `"white"`;

exports[`hsl(0, 5%, 85%) 1`] = `"white"`;

exports[`hsl(0, 10%, 0%) 1`] = `"black"`;

exports[`hsl(0, 10%, 5%) 1`] = `"black"`;

exports[`hsl(0, 10%, 10%) 1`] = `"red"`;

exports[`hsl(0, 10%, 15%) 1`] = `"red"`;

exports[`hsl(0, 10%, 20%) 1`] = `"red"`;

exports[`hsl(0, 10%, 25%) 1`] = `"red"`;

exports[`hsl(0, 10%, 30%) 1`] = `"red"`;

exports[`hsl(0, 10%, 35%) 1`] = `"red"`;

exports[`hsl(0, 10%, 40%) 1`] = `"red"`;

exports[`hsl(0, 10%, 45%) 1`] = `"red"`;

exports[`hsl(0, 10%, 50%) 1`] = `"red"`;

exports[`hsl(0, 10%, 55%) 1`] = `"red"`;

exports[`hsl(0, 10%, 60%) 1`] = `"red"`;

exports[`hsl(0, 10%, 65%) 1`] = `"todo"`;

exports[`hsl(0, 10%, 70%) 1`] = `"todo"`;

exports[`hsl(0, 10%, 75%) 1`] = `"todo"`;

exports[`hsl(0, 10%, 80%) 1`] = `"todo"`;

exports[`hsl(0, 10%, 85%) 1`] = `"white"`;

exports[`hsl(0, 15%, 0%) 1`] = `"black"`;

exports[`hsl(0, 15%, 5%) 1`] = `"black"`;

exports[`hsl(0, 15%, 10%) 1`] = `"red"`;

exports[`hsl(0, 15%, 15%) 1`] = `"red"`;

exports[`hsl(0, 15%, 20%) 1`] = `"red"`;

exports[`hsl(0, 15%, 25%) 1`] = `"red"`;

exports[`hsl(0, 15%, 30%) 1`] = `"red"`;

exports[`hsl(0, 15%, 35%) 1`] = `"red"`;

exports[`hsl(0, 15%, 40%) 1`] = `"red"`;

exports[`hsl(0, 15%, 45%) 1`] = `"red"`;

exports[`hsl(0, 15%, 50%) 1`] = `"red"`;

exports[`hsl(0, 15%, 55%) 1`] = `"red"`;

exports[`hsl(0, 15%, 60%) 1`] = `"red"`;

exports[`hsl(0, 15%, 65%) 1`] = `"todo"`;

exports[`hsl(0, 15%, 70%) 1`] = `"todo"`;

exports[`hsl(0, 15%, 75%) 1`] = `"todo"`;

exports[`hsl(0, 15%, 80%) 1`] = `"todo"`;

exports[`hsl(0, 15%, 85%) 1`] = `"white"`;

exports[`hsl(0, 20%, 0%) 1`] = `"black"`;

exports[`hsl(0, 20%, 5%) 1`] = `"black"`;

exports[`hsl(0, 20%, 10%) 1`] = `"red"`;

exports[`hsl(0, 20%, 15%) 1`] = `"red"`;

exports[`hsl(0, 20%, 20%) 1`] = `"red"`;

exports[`hsl(0, 20%, 25%) 1`] = `"red"`;

exports[`hsl(0, 20%, 30%) 1`] = `"red"`;

exports[`hsl(0, 20%, 35%) 1`] = `"red"`;

exports[`hsl(0, 20%, 40%) 1`] = `"red"`;

exports[`hsl(0, 20%, 45%) 1`] = `"red"`;

exports[`hsl(0, 20%, 50%) 1`] = `"red"`;

exports[`hsl(0, 20%, 55%) 1`] = `"red"`;

exports[`hsl(0, 20%, 60%) 1`] = `"red"`;

exports[`hsl(0, 20%, 65%) 1`] = `"todo"`;

exports[`hsl(0, 20%, 70%) 1`] = `"todo"`;

exports[`hsl(0, 20%, 75%) 1`] = `"todo"`;

exports[`hsl(0, 20%, 80%) 1`] = `"todo"`;

exports[`hsl(0, 20%, 85%) 1`] = `"white"`;

exports[`hsl(0, 25%, 0%) 1`] = `"black"`;

exports[`hsl(0, 25%, 5%) 1`] = `"black"`;

exports[`hsl(0, 25%, 10%) 1`] = `"red"`;

exports[`hsl(0, 25%, 15%) 1`] = `"red"`;

exports[`hsl(0, 25%, 20%) 1`] = `"red"`;

exports[`hsl(0, 25%, 25%) 1`] = `"red"`;

exports[`hsl(0, 25%, 30%) 1`] = `"red"`;

exports[`hsl(0, 25%, 35%) 1`] = `"red"`;

exports[`hsl(0, 25%, 40%) 1`] = `"red"`;

exports[`hsl(0, 25%, 45%) 1`] = `"red"`;

exports[`hsl(0, 25%, 50%) 1`] = `"red"`;

exports[`hsl(0, 25%, 55%) 1`] = `"red"`;

exports[`hsl(0, 25%, 60%) 1`] = `"red"`;

exports[`hsl(0, 25%, 65%) 1`] = `"todo"`;

exports[`hsl(0, 25%, 70%) 1`] = `"todo"`;

exports[`hsl(0, 25%, 75%) 1`] = `"todo"`;

exports[`hsl(0, 25%, 80%) 1`] = `"todo"`;

exports[`hsl(0, 25%, 85%) 1`] = `"white"`;

exports[`hsl(0, 30%, 0%) 1`] = `"black"`;

exports[`hsl(0, 30%, 5%) 1`] = `"black"`;

exports[`hsl(0, 30%, 10%) 1`] = `"red"`;

exports[`hsl(0, 30%, 15%) 1`] = `"red"`;

exports[`hsl(0, 30%, 20%) 1`] = `"red"`;

exports[`hsl(0, 30%, 25%) 1`] = `"red"`;

exports[`hsl(0, 30%, 30%) 1`] = `"red"`;

exports[`hsl(0, 30%, 35%) 1`] = `"red"`;

exports[`hsl(0, 30%, 40%) 1`] = `"red"`;

exports[`hsl(0, 30%, 45%) 1`] = `"red"`;

exports[`hsl(0, 30%, 50%) 1`] = `"red"`;

exports[`hsl(0, 30%, 55%) 1`] = `"red"`;

exports[`hsl(0, 30%, 60%) 1`] = `"red"`;

exports[`hsl(0, 30%, 65%) 1`] = `"todo"`;

exports[`hsl(0, 30%, 70%) 1`] = `"todo"`;

exports[`hsl(0, 30%, 75%) 1`] = `"todo"`;

exports[`hsl(0, 30%, 80%) 1`] = `"todo"`;

exports[`hsl(0, 30%, 85%) 1`] = `"white"`;

exports[`hsl(0, 35%, 0%) 1`] = `"black"`;

exports[`hsl(0, 35%, 5%) 1`] = `"black"`;

exports[`hsl(0, 35%, 10%) 1`] = `"red"`;

exports[`hsl(0, 35%, 15%) 1`] = `"red"`;

exports[`hsl(0, 35%, 20%) 1`] = `"red"`;

exports[`hsl(0, 35%, 25%) 1`] = `"red"`;

exports[`hsl(0, 35%, 30%) 1`] = `"red"`;

exports[`hsl(0, 35%, 35%) 1`] = `"red"`;

exports[`hsl(0, 35%, 40%) 1`] = `"red"`;

exports[`hsl(0, 35%, 45%) 1`] = `"red"`;

exports[`hsl(0, 35%, 50%) 1`] = `"red"`;

exports[`hsl(0, 35%, 55%) 1`] = `"red"`;

exports[`hsl(0, 35%, 60%) 1`] = `"red"`;

exports[`hsl(0, 35%, 65%) 1`] = `"light pink"`;

exports[`hsl(0, 35%, 70%) 1`] = `"light pink"`;

exports[`hsl(0, 35%, 75%) 1`] = `"white"`;

exports[`hsl(0, 35%, 80%) 1`] = `"white"`;

exports[`hsl(0, 35%, 85%) 1`] = `"white"`;

exports[`hsl(0, 40%, 0%) 1`] = `"black"`;

exports[`hsl(0, 40%, 5%) 1`] = `"black"`;

exports[`hsl(0, 40%, 10%) 1`] = `"red"`;

exports[`hsl(0, 40%, 15%) 1`] = `"red"`;

exports[`hsl(0, 40%, 20%) 1`] = `"red"`;

exports[`hsl(0, 40%, 25%) 1`] = `"red"`;

exports[`hsl(0, 40%, 30%) 1`] = `"red"`;

exports[`hsl(0, 40%, 35%) 1`] = `"red"`;

exports[`hsl(0, 40%, 40%) 1`] = `"red"`;

exports[`hsl(0, 40%, 45%) 1`] = `"red"`;

exports[`hsl(0, 40%, 50%) 1`] = `"red"`;

exports[`hsl(0, 40%, 55%) 1`] = `"red"`;

exports[`hsl(0, 40%, 60%) 1`] = `"red"`;

exports[`hsl(0, 40%, 65%) 1`] = `"light pink"`;

exports[`hsl(0, 40%, 70%) 1`] = `"light pink"`;

exports[`hsl(0, 40%, 75%) 1`] = `"white"`;

exports[`hsl(0, 40%, 80%) 1`] = `"white"`;

exports[`hsl(0, 40%, 85%) 1`] = `"white"`;

exports[`hsl(0, 45%, 0%) 1`] = `"black"`;

exports[`hsl(0, 45%, 5%) 1`] = `"black"`;

exports[`hsl(0, 45%, 10%) 1`] = `"red"`;

exports[`hsl(0, 45%, 15%) 1`] = `"red"`;

exports[`hsl(0, 45%, 20%) 1`] = `"red"`;

exports[`hsl(0, 45%, 25%) 1`] = `"red"`;

exports[`hsl(0, 45%, 30%) 1`] = `"red"`;

exports[`hsl(0, 45%, 35%) 1`] = `"red"`;

exports[`hsl(0, 45%, 40%) 1`] = `"red"`;

exports[`hsl(0, 45%, 45%) 1`] = `"red"`;

exports[`hsl(0, 45%, 50%) 1`] = `"red"`;

exports[`hsl(0, 45%, 55%) 1`] = `"red"`;

exports[`hsl(0, 45%, 60%) 1`] = `"red"`;

exports[`hsl(0, 45%, 65%) 1`] = `"light pink"`;

exports[`hsl(0, 45%, 70%) 1`] = `"light pink"`;

exports[`hsl(0, 45%, 75%) 1`] = `"white"`;

exports[`hsl(0, 45%, 80%) 1`] = `"white"`;

exports[`hsl(0, 45%, 85%) 1`] = `"white"`;

exports[`hsl(0, 50%, 0%) 1`] = `"black"`;

exports[`hsl(0, 50%, 5%) 1`] = `"black"`;

exports[`hsl(0, 50%, 10%) 1`] = `"red"`;

exports[`hsl(0, 50%, 15%) 1`] = `"red"`;

exports[`hsl(0, 50%, 20%) 1`] = `"red"`;

exports[`hsl(0, 50%, 25%) 1`] = `"red"`;

exports[`hsl(0, 50%, 30%) 1`] = `"red"`;

exports[`hsl(0, 50%, 35%) 1`] = `"red"`;

exports[`hsl(0, 50%, 40%) 1`] = `"red"`;

exports[`hsl(0, 50%, 45%) 1`] = `"red"`;

exports[`hsl(0, 50%, 50%) 1`] = `"red"`;

exports[`hsl(0, 50%, 55%) 1`] = `"red"`;

exports[`hsl(0, 50%, 60%) 1`] = `"red"`;

exports[`hsl(0, 50%, 65%) 1`] = `"light pink"`;

exports[`hsl(0, 50%, 70%) 1`] = `"light pink"`;

exports[`hsl(0, 50%, 75%) 1`] = `"white"`;

exports[`hsl(0, 50%, 80%) 1`] = `"white"`;

exports[`hsl(0, 50%, 85%) 1`] = `"white"`;

exports[`hsl(0, 55%, 0%) 1`] = `"black"`;

exports[`hsl(0, 55%, 5%) 1`] = `"black"`;

exports[`hsl(0, 55%, 10%) 1`] = `"red"`;

exports[`hsl(0, 55%, 15%) 1`] = `"red"`;

exports[`hsl(0, 55%, 20%) 1`] = `"red"`;

exports[`hsl(0, 55%, 25%) 1`] = `"red"`;

exports[`hsl(0, 55%, 30%) 1`] = `"red"`;

exports[`hsl(0, 55%, 35%) 1`] = `"red"`;

exports[`hsl(0, 55%, 40%) 1`] = `"red"`;

exports[`hsl(0, 55%, 45%) 1`] = `"red"`;

exports[`hsl(0, 55%, 50%) 1`] = `"red"`;

exports[`hsl(0, 55%, 55%) 1`] = `"red"`;

exports[`hsl(0, 55%, 60%) 1`] = `"red"`;

exports[`hsl(0, 55%, 65%) 1`] = `"light pink"`;

exports[`hsl(0, 55%, 70%) 1`] = `"light pink"`;

exports[`hsl(0, 55%, 75%) 1`] = `"white"`;

exports[`hsl(0, 55%, 80%) 1`] = `"white"`;

exports[`hsl(0, 55%, 85%) 1`] = `"white"`;

exports[`hsl(0, 60%, 0%) 1`] = `"black"`;

exports[`hsl(0, 60%, 5%) 1`] = `"black"`;

exports[`hsl(0, 60%, 10%) 1`] = `"red"`;

exports[`hsl(0, 60%, 15%) 1`] = `"red"`;

exports[`hsl(0, 60%, 20%) 1`] = `"red"`;

exports[`hsl(0, 60%, 25%) 1`] = `"red"`;

exports[`hsl(0, 60%, 30%) 1`] = `"red"`;

exports[`hsl(0, 60%, 35%) 1`] = `"red"`;

exports[`hsl(0, 60%, 40%) 1`] = `"red"`;

exports[`hsl(0, 60%, 45%) 1`] = `"red"`;

exports[`hsl(0, 60%, 50%) 1`] = `"red"`;

exports[`hsl(0, 60%, 55%) 1`] = `"red"`;

exports[`hsl(0, 60%, 60%) 1`] = `"red"`;

exports[`hsl(0, 60%, 65%) 1`] = `"pink"`;

exports[`hsl(0, 60%, 70%) 1`] = `"pink"`;

exports[`hsl(0, 60%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 60%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 60%, 85%) 1`] = `"white"`;

exports[`hsl(0, 65%, 0%) 1`] = `"black"`;

exports[`hsl(0, 65%, 5%) 1`] = `"black"`;

exports[`hsl(0, 65%, 10%) 1`] = `"red"`;

exports[`hsl(0, 65%, 15%) 1`] = `"red"`;

exports[`hsl(0, 65%, 20%) 1`] = `"red"`;

exports[`hsl(0, 65%, 25%) 1`] = `"red"`;

exports[`hsl(0, 65%, 30%) 1`] = `"red"`;

exports[`hsl(0, 65%, 35%) 1`] = `"red"`;

exports[`hsl(0, 65%, 40%) 1`] = `"red"`;

exports[`hsl(0, 65%, 45%) 1`] = `"red"`;

exports[`hsl(0, 65%, 50%) 1`] = `"red"`;

exports[`hsl(0, 65%, 55%) 1`] = `"red"`;

exports[`hsl(0, 65%, 60%) 1`] = `"red"`;

exports[`hsl(0, 65%, 65%) 1`] = `"pink"`;

exports[`hsl(0, 65%, 70%) 1`] = `"pink"`;

exports[`hsl(0, 65%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 65%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 65%, 85%) 1`] = `"white"`;

exports[`hsl(0, 70%, 0%) 1`] = `"black"`;

exports[`hsl(0, 70%, 5%) 1`] = `"black"`;

exports[`hsl(0, 70%, 10%) 1`] = `"red"`;

exports[`hsl(0, 70%, 15%) 1`] = `"red"`;

exports[`hsl(0, 70%, 20%) 1`] = `"red"`;

exports[`hsl(0, 70%, 25%) 1`] = `"red"`;

exports[`hsl(0, 70%, 30%) 1`] = `"red"`;

exports[`hsl(0, 70%, 35%) 1`] = `"red"`;

exports[`hsl(0, 70%, 40%) 1`] = `"red"`;

exports[`hsl(0, 70%, 45%) 1`] = `"red"`;

exports[`hsl(0, 70%, 50%) 1`] = `"red"`;

exports[`hsl(0, 70%, 55%) 1`] = `"red"`;

exports[`hsl(0, 70%, 60%) 1`] = `"red"`;

exports[`hsl(0, 70%, 65%) 1`] = `"pink"`;

exports[`hsl(0, 70%, 70%) 1`] = `"pink"`;

exports[`hsl(0, 70%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 70%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 70%, 85%) 1`] = `"white"`;

exports[`hsl(0, 75%, 0%) 1`] = `"black"`;

exports[`hsl(0, 75%, 5%) 1`] = `"black"`;

exports[`hsl(0, 75%, 10%) 1`] = `"red"`;

exports[`hsl(0, 75%, 15%) 1`] = `"red"`;

exports[`hsl(0, 75%, 20%) 1`] = `"red"`;

exports[`hsl(0, 75%, 25%) 1`] = `"red"`;

exports[`hsl(0, 75%, 30%) 1`] = `"red"`;

exports[`hsl(0, 75%, 35%) 1`] = `"red"`;

exports[`hsl(0, 75%, 40%) 1`] = `"red"`;

exports[`hsl(0, 75%, 45%) 1`] = `"red"`;

exports[`hsl(0, 75%, 50%) 1`] = `"red"`;

exports[`hsl(0, 75%, 55%) 1`] = `"red"`;

exports[`hsl(0, 75%, 60%) 1`] = `"red"`;

exports[`hsl(0, 75%, 65%) 1`] = `"pink"`;

exports[`hsl(0, 75%, 70%) 1`] = `"pink"`;

exports[`hsl(0, 75%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 75%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 75%, 85%) 1`] = `"white"`;

exports[`hsl(0, 80%, 0%) 1`] = `"black"`;

exports[`hsl(0, 80%, 5%) 1`] = `"black"`;

exports[`hsl(0, 80%, 10%) 1`] = `"red"`;

exports[`hsl(0, 80%, 15%) 1`] = `"red"`;

exports[`hsl(0, 80%, 20%) 1`] = `"red"`;

exports[`hsl(0, 80%, 25%) 1`] = `"red"`;

exports[`hsl(0, 80%, 30%) 1`] = `"red"`;

exports[`hsl(0, 80%, 35%) 1`] = `"red"`;

exports[`hsl(0, 80%, 40%) 1`] = `"red"`;

exports[`hsl(0, 80%, 45%) 1`] = `"red"`;

exports[`hsl(0, 80%, 50%) 1`] = `"red"`;

exports[`hsl(0, 80%, 55%) 1`] = `"red"`;

exports[`hsl(0, 80%, 60%) 1`] = `"red"`;

exports[`hsl(0, 80%, 65%) 1`] = `"pink"`;

exports[`hsl(0, 80%, 70%) 1`] = `"pink"`;

exports[`hsl(0, 80%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 80%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 80%, 85%) 1`] = `"white"`;

exports[`hsl(0, 85%, 0%) 1`] = `"black"`;

exports[`hsl(0, 85%, 5%) 1`] = `"black"`;

exports[`hsl(0, 85%, 10%) 1`] = `"red"`;

exports[`hsl(0, 85%, 15%) 1`] = `"red"`;

exports[`hsl(0, 85%, 20%) 1`] = `"red"`;

exports[`hsl(0, 85%, 25%) 1`] = `"red"`;

exports[`hsl(0, 85%, 30%) 1`] = `"red"`;

exports[`hsl(0, 85%, 35%) 1`] = `"red"`;

exports[`hsl(0, 85%, 40%) 1`] = `"red"`;

exports[`hsl(0, 85%, 45%) 1`] = `"red"`;

exports[`hsl(0, 85%, 50%) 1`] = `"red"`;

exports[`hsl(0, 85%, 55%) 1`] = `"red"`;

exports[`hsl(0, 85%, 60%) 1`] = `"red"`;

exports[`hsl(0, 85%, 65%) 1`] = `"red"`;

exports[`hsl(0, 85%, 70%) 1`] = `"red"`;

exports[`hsl(0, 85%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 85%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 85%, 85%) 1`] = `"white"`;

exports[`hsl(0, 90%, 0%) 1`] = `"black"`;

exports[`hsl(0, 90%, 5%) 1`] = `"black"`;

exports[`hsl(0, 90%, 10%) 1`] = `"red"`;

exports[`hsl(0, 90%, 15%) 1`] = `"red"`;

exports[`hsl(0, 90%, 20%) 1`] = `"red"`;

exports[`hsl(0, 90%, 25%) 1`] = `"red"`;

exports[`hsl(0, 90%, 30%) 1`] = `"red"`;

exports[`hsl(0, 90%, 35%) 1`] = `"red"`;

exports[`hsl(0, 90%, 40%) 1`] = `"red"`;

exports[`hsl(0, 90%, 45%) 1`] = `"red"`;

exports[`hsl(0, 90%, 50%) 1`] = `"red"`;

exports[`hsl(0, 90%, 55%) 1`] = `"red"`;

exports[`hsl(0, 90%, 60%) 1`] = `"red"`;

exports[`hsl(0, 90%, 65%) 1`] = `"red"`;

exports[`hsl(0, 90%, 70%) 1`] = `"red"`;

exports[`hsl(0, 90%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 90%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 90%, 85%) 1`] = `"white"`;

exports[`hsl(0, 95%, 0%) 1`] = `"black"`;

exports[`hsl(0, 95%, 5%) 1`] = `"black"`;

exports[`hsl(0, 95%, 10%) 1`] = `"red"`;

exports[`hsl(0, 95%, 15%) 1`] = `"red"`;

exports[`hsl(0, 95%, 20%) 1`] = `"red"`;

exports[`hsl(0, 95%, 25%) 1`] = `"red"`;

exports[`hsl(0, 95%, 30%) 1`] = `"red"`;

exports[`hsl(0, 95%, 35%) 1`] = `"red"`;

exports[`hsl(0, 95%, 40%) 1`] = `"red"`;

exports[`hsl(0, 95%, 45%) 1`] = `"red"`;

exports[`hsl(0, 95%, 50%) 1`] = `"red"`;

exports[`hsl(0, 95%, 55%) 1`] = `"red"`;

exports[`hsl(0, 95%, 60%) 1`] = `"red"`;

exports[`hsl(0, 95%, 65%) 1`] = `"red"`;

exports[`hsl(0, 95%, 70%) 1`] = `"red"`;

exports[`hsl(0, 95%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 95%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 95%, 85%) 1`] = `"white"`;

exports[`hsl(0, 100%, 0%) 1`] = `"black"`;

exports[`hsl(0, 100%, 5%) 1`] = `"black"`;

exports[`hsl(0, 100%, 10%) 1`] = `"red"`;

exports[`hsl(0, 100%, 15%) 1`] = `"red"`;

exports[`hsl(0, 100%, 20%) 1`] = `"red"`;

exports[`hsl(0, 100%, 25%) 1`] = `"red"`;

exports[`hsl(0, 100%, 30%) 1`] = `"red"`;

exports[`hsl(0, 100%, 35%) 1`] = `"red"`;

exports[`hsl(0, 100%, 40%) 1`] = `"red"`;

exports[`hsl(0, 100%, 45%) 1`] = `"red"`;

exports[`hsl(0, 100%, 50%) 1`] = `"red"`;

exports[`hsl(0, 100%, 55%) 1`] = `"red"`;

exports[`hsl(0, 100%, 60%) 1`] = `"red"`;

exports[`hsl(0, 100%, 65%) 1`] = `"red"`;

exports[`hsl(0, 100%, 70%) 1`] = `"red"`;

exports[`hsl(0, 100%, 75%) 1`] = `"light pink"`;

exports[`hsl(0, 100%, 80%) 1`] = `"light pink"`;

exports[`hsl(0, 100%, 85%) 1`] = `"white"`;
